src/client/session/mod.rs

```diff
@@ -31,21 +32,28 @@ lazy_static! {
 /// Session to be used with client operations. This acts as a handle to a server session.
 /// This keeps the details of how server sessions are pooled opaque to users.
 #[derive(Debug)]
-pub(crate) struct ClientSession {
+pub struct ClientSession {
     cluster_time: Option<ClusterTime>,
     server_session: ServerSession,
     client: Client,
     is_implicit: bool,
+    options: Option<SessionOptions>,
 }
 
 impl ClientSession {
```

Comment:
From the specification, it looks like we need to make a few of these methods public, namely:
- an accessor for the client
- `id()`
- `cluster_time()`
- `advance_cluster_time()`